File: 231. Power of Two/powerofTwo.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        #Set i at 0, use a while loop
        i = 0
        #As soon we find it's a power of two, return True
        while i < 31:
            if 2**i == n:
                return True
            #Else, increment i by one
            i += 1
        return False 
       
        

File: 231. Power of Two/powerofTwo.c
bool isPowerOfTwo(int n){
    //Take care of edge case, if the number is less than or equal to 0 return false
    if (n <= 0) return false;
    
    //If not, return not n&n-1
    return !(n&(n-1));
    
    //Explanation:
    //In binary, all powers of two will only contain one 1
    //For example, 4 is represented as 0100. 
    //0100 -1 flips the digits to 1011
    //0100 & 1011 == 0000 due to propositional logic
    //!0000 = 1111 == true, else false

}
